pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/NullHandlingIntegrationTest.java

```diff
@@ -326,6 +326,31 @@ public void testAggregateServerReturnFinalResult(boolean useMultiStageQueryEngin
     assertTrue(response.get("resultTable").get("rows").get(0).get(0).isNull());
   }
 
+  @Test
+  public void testWindowFunctionIgnoreNulls()
+      throws Exception {
+    // Window functions are only supported in the multi-stage query engine
+    setUseMultiStageQueryEngine(true);
+    String sqlQuery =
+        "SELECT salary, LAST_VALUE(salary) IGNORE NULLS OVER (ORDER BY DaysSinceEpoch) AS gapfilledSalary from "
```

Comment:
Does it work with bounded preceding/following? I remember running into some exception when trying it out

Reply:
Yes, it works with bounded preceding / following as well. 

> I remember running into some exception when trying it out

Was it something like this error during query planning:

```
Caused by: java.lang.RuntimeException: Failed to convert query to relational expression:
...
Caused by: java.lang.AssertionError: Conversion to relational algebra failed to preserve datatypes:
```

Interestingly, it looks like Calcite throws this error if the window function's input column is not nullable in the table schema (i.e., if `enableColumnBasedNullHandling` is `false` or the column has `"notNull": true`) and `IGNORE NULLS` or `RESPECT NULLS` option is used. While the error message is not the most clear, I don't think this is an actual bug because it doesn't really make sense to use those null handling related options if the window function's input column is not nullable.

Reply:
I guess we want to document it. Users might enable table level nullability (v1 engine nullability) and expect v2 engine to pick it up

Reply:
Yeah, good point, I'll make sure to add a note about this to the documentation. I plan to raise one consolidated documentation PR with changes from https://github.com/apache/pinot/pull/14273 and here.

Reply:
I don't think we should delegate this into something we document. We should also report the issue in Calcite Jira/email list and/or create a PR to fix it ourselfs. Same happened with the reserved keyword PR.